tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions/default.dart

```diff
@@ -66,4 +74,114 @@ class DefaultValueRestriction extends ValueRestriction {
 
     return errors;
   }
+
+  List<SourceSpanSeverityException> _booleanValidation(
+    dynamic value,
+    SourceSpan? span,
+  ) {
+    if (value is bool) return [];
+
+    var errors = <SourceSpanSeverityException>[];
+
+    if (value is! String ||
+        (value != defaultBooleanTrue && value != defaultBooleanFalse)) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid boolean: "true" or "false"',
+          span,
+        ),
+      );
+    }
+
+    return errors;
+  }
+
+  List<SourceSpanSeverityException> _integerValidation(
+    dynamic value,
+    SourceSpan? span,
+  ) {
+    if (value is int) return [];
+
+    var errors = <SourceSpanSeverityException>[];
+
+    if (value is! String) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid integer (e.g., "$key"=10).',
+          span,
+        ),
+      );
+    }
+
+    int? parsedValue = int.tryParse(value);
+    if (parsedValue == null) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid integer (e.g., "$key"=10).',
+          span,
+        ),
+      );
+    }
+    return errors;
+  }
+
+  List<SourceSpanSeverityException> _doubleValidation(
+    dynamic value,
+    SourceSpan? span,
+  ) {
+    if (value is double) return [];
+
+    var errors = <SourceSpanSeverityException>[];
+
+    if (value is! String) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid double (e.g., "$key"=10.5).',
+          span,
+        ),
+      );
+    }
+
+    double? parsedValue = double.tryParse(value);
+    if (parsedValue == null) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid double (e.g., "$key"=10.5).',
+          span,
+        ),
+      );
+    }
+    return errors;
+  }
+
+  List<SourceSpanSeverityException> _stringValidation(
+    dynamic value,
+    SourceSpan? span,
+  ) {
+    var errors = <SourceSpanSeverityException>[];
+
+    if (value is! String) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a valid string (e.g., "$key"=\'This is a string\' or "$key"="This is a string").',
+          span,
+        ),
+      );
+      return errors;
+    }
+
+    bool validDoubleQuote = RegExp(r'^"(\\.|[^"\\])*"$').hasMatch(value);
+    bool validSingleQuote = RegExp(r"^'(\\.|[^'\\])*'$").hasMatch(value);
+
+    if (!validDoubleQuote && !validSingleQuote) {
+      errors.add(
+        SourceSpanSeverityException(
+          'The "$key" value must be a properly quoted string (e.g., "$key"=\'This is a string\' or "$key"="This is a string").',
```

Comment:
Suggestion: Describe the issue a bit better.

I don't understand what has gone wrong in the string here, please improve this error message.